src/main/java/mate/academy/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package mate.academy.bookstore.service.impl;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.cartitem.AddToCartRequestDto;
+import mate.academy.bookstore.dto.shoppingcarts.ShoppingCartDto;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import mate.academy.bookstore.mapper.ShoppingCartMapper;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.model.CartItem;
+import mate.academy.bookstore.model.ShoppingCart;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.book.BookRepository;
+import mate.academy.bookstore.repository.cartitem.CartItemRepository;
+import mate.academy.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.bookstore.repository.user.UserRepository;
+import mate.academy.bookstore.service.ShoppingCartService;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public ShoppingCartDto getById(Long id) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Can't find shopping cart by id = " + id));
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto addToCart(AddToCartRequestDto requestDto, Long userId) {
+        User user = userRepository.findById(userId).orElseThrow(() ->
+                new EntityNotFoundException("Can't find user by id = " + userId));
+        Book book = bookRepository.findById(requestDto.bookId()).orElseThrow(() ->
+                new EntityNotFoundException("Can't find book by id = " + requestDto.bookId()));
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId).orElseGet(() -> {
```

Comment:
better extract it to a separate method (for example: ShoppingCart shoppingCart = getOrCreateShoppingCart())